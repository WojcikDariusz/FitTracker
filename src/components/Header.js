import React, { Component } from 'react';

class Header extends Component {

  constructor() {
    super() 
      this.state = {
        nameOfTracker : "I AM THE BEST",
        clicked : true,
        textColor : "black",
        backgroundColor : "white"
    }
  }

handleClick = () => {

  if(this.state.clicked === true) {
    this.setState({
      nameOfTracker : "BEST TRACKING APP",
      textColor : "white",
      backgroundColor : "black"
    })
  } else {
    this.setState({
      nameOfTracker: "I AM THE BEST",
      textColor : "black",
      backgroundColor : "white"
    })
  }
  this.setState({
    clicked : !this.state.clicked
  })
};


    render() {

      let headerCss = {

        color : this.state.textColor,
        backgroundColor : this.state.backgroundColor
      
      }
      

        return (
        <div className='header' style={headerCss} onClick={this.handleClick}>
            <center>
             <h2>{this.state.nameOfTracker}</h2>
            </center>
        </div>
        )
    };
};
export default Header;