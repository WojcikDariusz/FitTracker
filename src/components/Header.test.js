import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header';
import {configure,shallow} from 'enzyme';
import adapter from 'enzyme-adapter-react-16';

configure({adapter: new adapter()});

describe('Header tests', () => {

    it('Correctly renders the header html', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Header />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('Renders the text', () => {
        const wrapper = shallow(<Header />);
        expect(wrapper.find('h2').text()).toBe('I AM THE BEST');
        wrapper.setState({
            nameOfTracker : "BEST TRACKING APP"
        });
        expect(wrapper.find('h2').text()).toBe('BEST TRACKING APP');
    });

    it('Checking if handleClick method works correctly, simulate changes after clicking Header DIV', () => {
        const wrapper = shallow(<Header />);
        expect(wrapper.state().nameOfTracker).toBe('I AM THE BEST');
        
        wrapper.find('.header').simulate('click');
        expect(wrapper.state().nameOfTracker).toBe('BEST TRACKING APP')
    });

    it('Snapshot matches', () => {
        const wrapper = shallow(<Header />);
        expect(wrapper).toMatchSnapshot();
    });
});