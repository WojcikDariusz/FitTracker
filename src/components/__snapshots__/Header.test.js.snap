// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests Snapshot matches 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <center>
        <h2>
          I AM THE BEST
        </h2>
      </center>,
      "className": "header",
      "onClick": [Function],
      "style": Object {
        "backgroundColor": "white",
        "color": "black",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h2>
          I AM THE BEST
        </h2>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "I AM THE BEST",
        },
        "ref": null,
        "rendered": "I AM THE BEST",
        "type": "h2",
      },
      "type": "center",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <center>
          <h2>
            I AM THE BEST
          </h2>
        </center>,
        "className": "header",
        "onClick": [Function],
        "style": Object {
          "backgroundColor": "white",
          "color": "black",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h2>
            I AM THE BEST
          </h2>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "I AM THE BEST",
          },
          "ref": null,
          "rendered": "I AM THE BEST",
          "type": "h2",
        },
        "type": "center",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
